### ISSUE TITLE
Enhancement: consistent-type-imports support `importKind` type

### ISSUE THREAD

**Issue by kentcdodds** [typescript-eslint/typescript-eslint#5927]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-type-imports/

### Description

I'd like an option that allows ESLint to auto-fix from this:

```
import type { ActionArgs } from '@remix-run/node'
import { json } from '@remix-run/node'
```

To this:

```
import { type ActionArgs, json } from '@remix-run/node'
```

Basically, I want to only have a single import from a single module *and* I want type imports to be treated as type imports.

### Fail

```typescript
import type { ActionArgs } from '@remix-run/node'
```
```


### Pass

```typescript
import { type ActionArgs } from '@remix-run/node'
```
```


### Additional Info

_No response_

**Comment by Josh-Cena** [comment]
Looks like a duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/4338?

**Comment by bradzacher** [comment]
Yup duplicate!

**Comment by kentcdodds** [comment]
Thanks! Sorry I missed it.

**Comment by snewcomer** [comment]
@kentcdodds We are close!  2 more PRs left❗❕ 

https://github.com/import-js/eslint-plugin-import/pull/2475
https://github.com/typescript-eslint/typescript-eslint/pull/5050
